synapse/util/caches/descriptors.py

```diff
@@ -220,7 +220,9 @@ def __init__(
         self.iterable = iterable
         self.prune_unread_entries = prune_unread_entries
 
-    def __get__(self, obj: Optional[Any], owner: Optional[Type]) -> Callable[..., Any]:
+    def __get__(
+        self, obj: Optional[Any], owner: Optional[Type]
+    ) -> Callable[..., "defer.Deferred[Any]"]:
```

Comment:
Is this the method that the plugin changes target?

Reply:
Kind of? The plugin doesn't really care about any of the internals of the descriptors, but I believe this corresponds to what I'm doing.